docs/proposals/federation.md

```diff
@@ -404,36 +494,107 @@ These are often left implicit by customers, but are worth calling out explicitly
+   independently (different sets of developers, different release
+   schedules etc).
+4. Administration complexity: Again, I think that this could be argued
+   both ways.  Superficially it woud seem that administration of a
+   single Monolithic multi-zone cluster might be simpler by virtue of
+   being only "one thing to manage", however in practise each of the
+   underlying availability zones (and possibly cloud providers) has
+   it's own capacity, pricing, hardware platforms, and possibly
+   bureaucratic boudaries (e.g. "our EMEA IT department manages those
+   European clusters").  So explicitly allowing for (but not
+   mandating) completely independent administration of each underlying
+   Kubernetes cluster, and the Federation system itself, in the
+   Decoupled Hierarchical model seems to have real practical benefits
+   that outweigh the superficial simplicity of the Monolithic model.
+5. Application development and deployment complexity:  It's not clear
+   to me that there is any significant difference between the two
+   models in this regard.  Presumably the API exposed by the two
+   different architectures would look very similar, as would the
+   behavior of the deployed applications.
+
+### The Decoupled, Hierarchical Approach - Illustrated
 
 ![image](federation-high-level-arch.png)
 
 ## Ubernetes API
 
-This looks a lot like the existing Kubernetes API but is explicitly multi-cluster.
+It is proposed that this look a lot like the existing Kubernetes API
+but be explicitly multi-cluster.
 
-+  Clusters become first class objects, which can be registered, listed, described, deregistered etc via the API.
-+  Compute resources can be explicitly requested in specific clusters, or automatically scheduled to the "best" cluster by Ubernetes (by a pluggable Policy Engine).
-+  There is a federated equivalent of a replication controller type, which is multicluster-aware, and delegates to cluster-specific replication controllers as required (e.g. a federated RC for n replicas might simply spawn multiple replication controllers in different clusters to do the hard work).
-+ These federated replication controllers (and in fact all the
-   services comprising the Ubernetes Control Plane) have to run
-   somewhere.  For high availability Ubernetes deployments, these
-   services may run in a dedicated Kubernetes cluster, not physically
-   co-located with any of the federated clusters.  But for simpler
-   deployments, they may be run in one of the federated clusters (but
-   when that cluster goes down, Ubernetes is down, obviously).
++ Clusters become first class objects, which can be registered,
+   listed, described, deregistered etc via the API.
++ Compute resources can be explicitly requested in specific clusters,
+   or automatically scheduled to the "best" cluster by Ubernetes (by a
+   pluggable Policy Engine).
++ There is a federated equivalent of a replication controller type,
```

Comment:
Deployment, probably, instead of RC.
https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/deployment.md